server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreLockManagerFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.store.lockmanager;
+
+import org.opensearch.common.blobstore.BlobContainer;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.index.store.RemoteBufferedOutputDirectory;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.repositories.Repository;
+import org.opensearch.repositories.RepositoryMissingException;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+import org.opensearch.index.store.RemoteDirectory;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+/**
+ * Factory for remote store lock manager
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreLockManagerFactory {
+    private static final String SEGMENTS = "segments";
+    private static final String LOCK_FILES = "lock_files";
+    private final Supplier<RepositoriesService> repositoriesService;
+    public RemoteStoreLockManagerFactory(Supplier<RepositoriesService> repositoriesService) {
+        this.repositoriesService = repositoriesService;
+    }
+
+    public RemoteStoreMetadataLockManager newLockManager(String repositoryName, String indexUUID,
+                                                        String shardId) throws IOException {
+        return newLockManager(repositoriesService.get(), repositoryName, indexUUID, shardId);
+    }
+
+    public static RemoteStoreMetadataLockManager newLockManager(RepositoriesService repositoriesService,
+                                                               String repositoryName,
+                                                               String indexUUID,
+                                                               String shardId) throws IOException {
+        try (Repository repository = repositoriesService.repository(repositoryName)) {
+            assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
+            BlobPath shardLevelBlobPath = ((BlobStoreRepository) repository).basePath().add(indexUUID)
+                .add(shardId).add(SEGMENTS);
+            RemoteDirectory shardMDLockDirectory = createRemoteBufferedOutputDirectory(repository, shardLevelBlobPath,
```

Comment:
IMO, `lock_files` can remain at `shardId` level. Nesting it within `segments` breaks the assumption that directory does not have files in nested format (though I don't think it will break any functionality).